Testing/UnitTesting/Hdf5RecorderTests.cpp

```diff
@@ -11,17 +11,26 @@ TEST(Hdf5RecorderTest, CreateInstanceSuccess)
    ASSERT_TRUE(recorder != nullptr);
 }
 
-// Test case for open file successfully
-TEST(Hdf5RecorderTest, Hdf5InitTest)
+// Test case for init() and term()
+TEST(Hdf5RecorderTest, Hdf5InitAndTermTest)
 {
-   // Create an instance of Hdf5Recorder
-   std::string outputFile = "../Testing/UnitTesting/TestOutput/Hdf5test_output.h5";
+   // Create an instance of Hdf5Recorder with a specific output file name
+   std::string outputFile = "../Testing/UnitTesting/TestOutput/Hdf5test_output_term.h5";
    Hdf5Recorder recorder(outputFile);
-   /*recorder.init();
-   // Test to see if output file exist
-   FILE *f = fopen("../Testing/UnitTesting/TestOutput/Hdf5test_output.h5", "r");
+   recorder.init();
```

Comment:
Just a comment (you'll need to resolve it to re-request my review): let's talk at our next meeting about where `init()` gets called from the simulator code. I did a quick scan of the XML recorder code and didn't see such a call, but clearly it should happen. I just want to double check.